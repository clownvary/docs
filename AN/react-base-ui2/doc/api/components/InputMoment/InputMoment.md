## InputMoment  extends InputBase
---
InputMoment component allows you validate and enter manually date and time.

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
value|moment &#124; Date &#124; string|null|false|Gets or sets the date value for a date input.
min|moment &#124; Date|undefined|false|Determines the minimal date that can be entered.
max|moment &#124; Date|undefined|false|Determines the maximum date that can be entered.
format|string|'d'|false|The format pattern to display the date value.<br>InputMoment supports two types of formats: Standard Format and Custom Format.<br>A standard date and time format string uses a single format specifier to  define the text representation of a date and time value.<br> Possible values for Standard Format are:<br> - "d": ShortDatePattern  M/D/YYYY  - "D": LongDatePattern  dddd, MMMM DD, YYYY  - "f": Full date and time (long date and short time)  dddd, MMMM DD, YYYY h:mm A  - "F": Full date and time (long date and long time) dddd, MMMM DD, YYYY h:mm:ss A  - "g": General (short date and short time) M/d/YYYY h:mm A  - "G": General (short date and long time) M/d/YYYY h:mm:ss A  - "m": month/day pattern MMMM DD  - "M": month/day pattern MMMM DD  - "s": sortable format that does not vary by culture  -      YYYY\u0027-\u0027MM\u0027-\u0027dd\u0027T\u0027HH\u0027:\u0027mm\u0027:\u0027ss  - "t": short time pattern h:mm A  - "T": long time pattern h:mm:ss A  - "u": Universal Sortable DateTime Pattern, same to s  - "U": Full date and time (long date and long time) using universal time  -    same to F  - "y": month/year pattern YYYY MMMM  - "Y": month/year pattern YYYY MMMM<br> Any date and time format string that contains more than one character, including white space,  is interpreted as a custom date and time format string. For example:  "mmm-DD-YYYY", "mmmm d, YYYY", "mm/DD/YYYY", "d-mmm-YYYY", "ddd, mmmm DD, YYYY" etc.<br> Below are the custom date and time format specifiers:<br> - "D": The day of the month, from 1 through 31.  - "DD": The day of the month, from 01 through 31.  - "ddd": The abbreviated name of the day of the week.  - "dddd": The full name of the day of the week.  - "m": The minute, from 0 through 59.  - "mm": The minute, from 00 through 59.  - "M": The month, from 1 through 12.  - "MM": The month, from 01 through 12.  - "MMM": The abbreviated name of the month.  - "MMMM": The full name of the month.  - "Y": The year, from 0 to 99.  - "YY": The year, from 00 to 99  - "YYYY": The year as a four-digit number  - "h": The hour, using a 12-hour clock from 1 to 12.  - "hh": The hour, using a 12-hour clock from 01 to 12.  - "H": The hour, using a 24-hour clock from 0 to 23.  - "HH": The hour, using a 24-hour clock from 00 to 23.  - "s": The second, from 0 through 59.  - "ss": The second, from 00 through 59.  - "a": The am/pm designator.  - "A": The AM/PM designator.
allowKeySpin|bool|true|false|Allow spinning value by up/down key.
triggerIcon|string|'icon-calendar-m'|false|Icon class name for the 1st trigger button.
triggerIconToggle|string|''|false|Icon class name for the 1st trigger button in toggle state.
triggerIcon2|string|'icon-clock-m'|false|Icon class name for the 2nd trigger button.
triggerIconToggle2|string|''|false|Icon class name for the 2nd trigger button in toggle state.
timeStep|number|30|false|Step in minutes when generating the time picker list.
onCalendarOpen|func|identity|false|Callback function that will be called when the calendar is dropdown.
onCalendarClose|func|identity|false|Callback function that will be called when the calendar is closed.
onValueChange|func|identity|false|The onValueChange event handler.A function called when the value of the input is changed.