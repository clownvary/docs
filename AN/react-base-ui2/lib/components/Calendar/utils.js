'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _moment = require('moment');var _moment2 = _interopRequireDefault(_moment);
var _isArray = require('lodash/isArray');var _isArray2 = _interopRequireDefault(_isArray);
var _uniqWith = require('lodash/uniqWith');var _uniqWith2 = _interopRequireDefault(_uniqWith);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var getTenYearRange = function getTenYearRange(mmomentDate) {
  var year = mmomentDate.year();
  var begin = year - year % 10;
  var end = begin + 9;
  return begin + '-' + end;
};

var getWeeks = function getWeeks(currentDate, firstDayOfWeek) {
  var now = currentDate.clone();
  var veryShortWeekdays = [];
  var weekDays = [];
  var localeData = currentDate.localeData();
  for (var dateColIndex = 0; dateColIndex < 7; dateColIndex += 1) {
    var index = (firstDayOfWeek + dateColIndex) % 7;
    now.day(index);
    weekDays[dateColIndex] = localeData.weekdays(now);
    veryShortWeekdays[dateColIndex] = weekDays[dateColIndex].substring(0, 1);
  }

  return { veryShortWeekdays: veryShortWeekdays, weekDays: weekDays };
};

var compareByFormat = function compareByFormat(date1, date2, format) {return (
    date1.format(format) === date2.format(format));};

var getSafeValue = function getSafeValue(value) {
  if (!(0, _isArray2.default)(value)) {
    value = [value];
  }
  var safeValue = value.map(function (v) {return v && (0, _moment2.default)(v);}).filter(function (v) {return v && v.isValid();});

  return (0, _uniqWith2.default)(safeValue, function (m1, m2) {return m1.isSame(m2, 'day');});
};

var sortedDates = function sortedDates(value) {return value.map(function (d) {return d.clone();}).
  sort(function (a, b) {
    if (a.isAfter(b, 'day')) {
      return 1;
    }
    if (a.isBefore(b, 'day')) {
      return -1;
    }
    return 0;
  });};exports.default =

{
  getTenYearRange: getTenYearRange,
  getWeeks: getWeeks,
  compareByFormat: compareByFormat,
  getSafeValue: getSafeValue,
  sortedDates: sortedDates };module.exports = exports['default'];