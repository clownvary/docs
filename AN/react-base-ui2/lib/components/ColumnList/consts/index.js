'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _ColumnType = require('./ColumnType');var _ColumnType2 = _interopRequireDefault(_ColumnType);
var _SelectionMode = require('../../../consts/SelectionMode');var SelectionMode = _interopRequireWildcard(_SelectionMode);
var _SortOrder = require('../../../consts/SortOrder');var _SortOrder2 = _interopRequireDefault(_SortOrder);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

{
  SortOrder: _SortOrder2.default,
  ColumnType: _ColumnType2.default,
  SelectionMode: SelectionMode };module.exports = exports['default'];