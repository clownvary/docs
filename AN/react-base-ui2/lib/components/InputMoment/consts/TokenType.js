'use strict';Object.defineProperty(exports, "__esModule", { value: true }); /** Token Type
                                                                             * @enum {string}
                                                                             * @memberof InputMoment
                                                                            */
var TokenType = {
  LITERAL: 'literal',
  MONTH: 'M',
  MONTH_TWO_DIGITS: 'MM',
  MONTH_SHORT_NAME: 'MMM',
  MONTH_LONG_NAME: 'MMMM',
  DATE: 'D',
  DATE_TWO_DIGITS: 'DD',
  DAY_SHORT_NAME: 'ddd',
  DAY_LONG_NAME: 'dddd',
  YEAR: 'Y',
  YEAR_TWO_DIGITS: 'YY',
  YEAR_FOUR_DIGITS: 'YYYY',
  HOUTR_12: 'h',
  HOUTR_12_TWO_DIGITS: 'hh',
  HOUTR_24: 'H',
  HOUTR_24_TWO_DIGITS: 'HH',
  AMPM: 'a',
  AMPM_UPPER: 'A',
  MINUTE: 'm',
  MINUTE_TWO_DIGITS: 'mm',
  SECOND: 's',
  SECOND_TWO_DIGITS: 'ss' };exports.default =


TokenType;module.exports = exports['default'];