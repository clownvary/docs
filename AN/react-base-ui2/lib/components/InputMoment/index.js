'use strict';Object.defineProperty(exports, "__esModule", { value: true });exports.InputMomentProps = exports.InputMoment = undefined;var _InputMoment = require('./InputMoment');var _InputMoment2 = _interopRequireDefault(_InputMoment);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.


InputMoment = _InputMoment2.default;exports.
InputMomentProps = _InputMoment.InputMomentProps;exports.default =


_InputMoment2.default;