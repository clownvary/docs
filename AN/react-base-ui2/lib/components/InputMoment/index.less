/* stylelint-disable */
.an-input-calendar__flexible {
  .an-calendar-year,
  .an-calendar-month,
  .an-calendar-day {
    width: auto;
  }
}
