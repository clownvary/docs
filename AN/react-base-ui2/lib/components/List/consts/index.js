'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _ListType = require('./ListType');var _ListType2 = _interopRequireDefault(_ListType);
var _SelectionMode = require('../../../consts/SelectionMode');var SelectionMode = _interopRequireWildcard(_SelectionMode);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

{
  ListType: _ListType2.default,
  SelectionMode: SelectionMode };module.exports = exports['default'];