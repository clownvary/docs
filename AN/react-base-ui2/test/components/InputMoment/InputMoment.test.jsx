import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import InputMoment from '../../../src/components/InputMoment';
import { createMoment, isSame } from '../../../src/utils/momentHelper';
import { Globalize } from '../../../src/services/i18n';
import TokenType from '../../../src/components/InputMoment/consts/TokenType';
import { KeyCode } from '../../../src/consts';

const setup = (props = {}) => mount(<InputMoment {...props} />);

describe('components/InputMoment/InputMoment.jsx', () => {
  let wrapper;
  const m1 = createMoment('2018-01-01T00:00:00+00:00').utc();
  const m2 = createMoment('2018-01-02T00:00:00+00:00').utc();
  const m3 = createMoment('2018-01-03T00:00:00+00:00').utc();
  const t1 = '2018-01-01T00:00:00+00:00';
  const t2 = '2018-01-02T00:00:00+00:00';
  const t3 = '2018-01-03T00:00:00+00:00';
  const r1 = '01 Jan 2018';
  const r2 = '02 Jan 2018';
  const r3 = '03 Jan 2018';

  beforeEach(() => {
    const value = m1;
    wrapper = setup({ value });
  });

  test('rendering and basic useage', () => {
    const instance = wrapper.instance();
    expect(instance.isBlank()).toBe(false);
    expect(toJSON(wrapper)).toMatchSnapshot();

    instance.setValue('2018-01-01T00:00:00+00:00');
    expect(isSame(instance.getValue(), m1)).toBe(true);

    wrapper.setProps({ allowBlank: false });
    instance.onClear();
    expect(isSame(instance.getValue(), m1)).toBe(true);

    wrapper.setProps({ allowBlank: true });
    instance.onClear();
    expect(instance.getValue()).toBe(null);
  });

  test('get and set', () => {
    const instance = wrapper.instance();
    expect(isSame(instance.value, m1)).toBe(true);
    expect(instance.text).toBe(r1);

    instance.value = m2;
    expect(instance.text).toBe(r2);

    instance.text = t3;
    expect(isSame(instance.value, m3));
  });

  test('component lifecycs', () => {
    const instance = wrapper.instance();
    wrapper.setProps({ value: m2 });
    expect(instance.getText()).toBe(r2);

    wrapper.setProps({ max: m3 });
    expect(instance.needCheckRange).toBe(false);

    wrapper.setProps({ format: TokenType.YEAR_FOUR_DIGITS });
    expect(instance.getText()).toBe('2018');

    wrapper.setProps({ timeStep: 100 });
    expect(instance.listItems.length).toBeGreaterThan(0);
  });

  test('handle input events', () => {
    const instance = wrapper.instance();
    const input = wrapper.find('input');
    jest.spyOn(instance, 'setValue');
    jest.spyOn(instance, 'allowEdit');
    jest.spyOn(instance, 'updateText');
    jest.spyOn(instance, 'triggerEvent');
    jest.spyOn(instance.textProvider, 'isValid');
    jest.spyOn(instance.textProvider, 'getField');
    jest.spyOn(instance.textProvider, 'isFieldGap');
    jest.spyOn(instance.textProvider, 'closeBlankField');
    const dateNowSpy = jest.spyOn(Date, 'now').mockImplementation(() => 1527825600000);

    jest.useFakeTimers();

    input.simulate('focus');
    jest.runAllTimers();

    instance.allowEdit.mockImplementationOnce(() => false);
    expect(instance.onInputFocus()).toBeUndefined();

    input.simulate('blur');
    jest.runAllTimers();

    instance.onIMEBreakThrough();
    expect(instance.imeMode).toBe(true);

    instance.input.value = t1;
    instance.onStopTyping();
    expect(instance.setValue).toBeCalled();

    instance.textProvider.closeBlankField.mockImplementationOnce(() => true);
    instance.input.value = t2;
    instance.onStopTyping();
    expect(instance.updateText).toBeCalled();

    // onKeyPressPreview
    instance.textProvider.getField.mockImplementationOnce(() => null);
    expect(instance.onKeyPressPreview('1')).toBe(true);

    instance.textProvider.getField.mockImplementationOnce(() => ({ allowInstanceEditing: true }));
    expect(instance.onKeyPressPreview('a')).toBe(true);

    instance.textProvider.isValid.mockImplementationOnce(() => false);
    expect(instance.onKeyPressPreview('1')).toBe(true);

    expect(instance.onKeyPressPreview(' ')).toBe(true);
    instance.onKeyPressPreview('0');
    instance.onKeyPressPreview('1');
    expect(instance.getText()).toBe('01 Jun 2018');

    instance.textProvider.isFieldGap.mockImplementationOnce(() => true);
    instance.input.value = t1;
    instance.onKeyPressPreview('3');
    expect(instance.getText()).toBe(r1);

    instance.timeStamp = new Date((new Date()).getTime() - 100);
    instance.onKeyPressPreview('1');
    expect(instance.triggerEvent).toBeCalledWith('onInvalidInput', { char: '1' });

    // onKeyDownPreview - navigate between fields
    instance.input.value = t1;
    instance.toFirstField();
    instance.input.value = t1;
    instance.onKeyDownPreview({ keyCode: KeyCode.RIGHT });
    expect(instance.textProvider.activeFieldIndex).toBe(1);
    instance.input.value = t1;
    instance.onKeyDownPreview({ keyCode: KeyCode.LEFT });
    expect(instance.textProvider.activeFieldIndex).toBe(0);
    instance.input.value = t1;
    instance.onKeyDownPreview({ keyCode: KeyCode.TAB });
    expect(instance.textProvider.activeFieldIndex).toBe(1);
    instance.input.value = t1;
    instance.onKeyDownPreview({ keyCode: KeyCode.TAB, shiftKey: true });
    expect(instance.textProvider.activeFieldIndex).toBe(0);
    instance.input.value = t1;
    instance.onKeyDownPreview({ keyCode: KeyCode.TAB, shiftKey: true });
    expect(instance.textProvider.activeFieldIndex).toBe(0);
    instance.input.value = t1;
    instance.toLastField();
    instance.input.value = t1;
    instance.onKeyDownPreview({ keyCode: KeyCode.TAB });
    expect(instance.textProvider.activeFieldIndex).toBe(instance.textProvider.getFieldCount() - 1);
    instance.input.value = t1;
    instance.onKeyDownPreview({ keyCode: KeyCode.HOME });
    expect(instance.textProvider.activeFieldIndex).toBe(0);
    instance.input.value = t1;
    instance.onKeyDownPreview({ keyCode: KeyCode.END });
    expect(instance.textProvider.activeFieldIndex).toBe(instance.textProvider.getFieldCount() - 1);

    instance.input.value = t1;
    instance.toFirstField();
    instance.textProvider.isValid.mockImplementation(() => false);
    instance.input.value = t1;
    instance.onKeyDownPreview({ which: KeyCode.TAB });
    expect(instance.textProvider.activeFieldIndex).toBe(0);

    dateNowSpy.mockReset();
    dateNowSpy.mockRestore();
  });

  test('field operations', () => {
    let instance = wrapper.instance();

    jest.useFakeTimers();

    // highLightField(index)
    jest.spyOn(instance, 'isFocused');
    jest.spyOn(instance, 'select');
    jest.spyOn(instance.textProvider, 'getFieldRange');

    expect(instance.highLightField(-1)).toBeUndefined();
    expect(instance.isFocused).not.toBeCalled();

    instance.isFocused.mockImplementationOnce(() => true);
    instance.textProvider.getFieldRange.mockImplementationOnce(() => ({ start: 0, end: 5 }));
    instance.highLightField(0);
    jest.runAllTimers();
    expect(instance.select).toBeCalledWith({ start: 0, end: 5 });

    // move between fields
    instance.setActiveField(0);
    expect(instance.textProvider.activeFieldIndex).toBe(0);
    instance.toNextField();
    expect(instance.textProvider.activeFieldIndex).toBe(1);
    instance.toLastField();
    expect(instance.textProvider.activeFieldIndex).toBe(instance.textProvider.getFieldCount() - 1);
    instance.toPrevField();
    expect(instance.textProvider.activeFieldIndex).toBe(instance.textProvider.getFieldCount() - 2);
    instance.toFirstField();
    expect(instance.textProvider.activeFieldIndex).toBe(0);

    // highLightCurosr(pos)
    jest.spyOn(instance, 'highLightField');
    jest.spyOn(instance, 'setActiveField');
    instance.highLightCursor(0);
    expect(instance.highLightField).toHaveBeenCalledTimes(1);
    expect(instance.setActiveField).toHaveBeenCalledTimes(1);

    wrapper = setup({ format: '[LITERAL]' });
    instance = wrapper.instance();
    jest.spyOn(instance, 'allowEdit');
    jest.spyOn(instance, 'highLightField');
    jest.spyOn(instance, 'setActiveField');

    expect(instance.highLightCursor(0)).toBeUndefined();
    expect(instance.setActiveField).not.toBeCalled();
    expect(instance.highLightField).not.toBeCalled();

    instance.allowEdit.mockImplementationOnce(() => false);
    expect(instance.highLightCursor(0)).toBeUndefined();
    expect(instance.setActiveField).not.toBeCalled();
    expect(instance.highLightField).not.toBeCalled();

    instance.highLightCursor();
    expect(instance.setActiveField).not.toBeCalled();
    expect(instance.highLightField).not.toBeCalled();
  });

  test('deletion', () => {
    wrapper = setup({ value: m3 });
    let instance = wrapper.instance();
    jest.spyOn(instance, 'allowEdit');
    jest.spyOn(instance, 'isSelectAll');
    jest.spyOn(instance.textProvider, 'clearField');

    instance.allowEdit.mockImplementationOnce(() => false);
    expect(instance.deleteSelection()).toBeUndefined();
    expect(isSame(instance.getValue(), m3));

    instance.deleteSelection();
    expect(instance.getText()).toBe(r3.replace(/03/, ''));

    instance.isSelectAll.mockImplementationOnce(() => true);
    instance.deleteSelection();
    expect(instance.getValue()).toBe(null);

    wrapper.setProps({ allowBlank: false });
    instance.isSelectAll.mockImplementationOnce(() => true);
    expect(instance.deleteSelection()).toBeUndefined();

    wrapper = setup({ value: m3 });
    instance = wrapper.instance();
    jest.spyOn(instance.textProvider, 'clearField');
    instance.textProvider.clearField.mockImplementationOnce(() => false);
    instance.deleteSelection();
    expect(instance.getText()).toBe(r3);
  });

  test('handle calendar', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'onCalendarOpen');
    jest.spyOn(instance, 'popupCalendar');
    jest.spyOn(instance, 'triggerEvent');

    instance.onTriggerClick();
    expect(instance.popupCalendar).toHaveBeenCalledTimes(1);
    expect(instance.onCalendarOpen).toHaveBeenCalledTimes(1);

    instance.setValue(null);
    instance.onTriggerClick();
    expect(instance.popupCalendar).toHaveBeenCalledTimes(2);
    expect(instance.onCalendarOpen).toHaveBeenCalledTimes(2);

    instance.onCalendarClose();
    expect(instance.triggerEvent).toBeCalledWith('onCalendarClose');
  });

  test('handle list', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'onListOpen');
    jest.spyOn(instance, 'findInList');
    instance.onTrigger2Click();
    expect(instance.onListOpen).toBeCalled();

    instance.value = null;
    instance.onTrigger2Click();
    expect(instance.findInList).toBeCalled();

    expect(instance.onListSelected()).toBeUndefined();
    expect(instance.onListSelected([-1])).toBeUndefined();
    expect(instance.onListSelected([instance.listItems.length])).toBeUndefined();

    instance.preMoment = m1;
    instance.onListSelected([1]);
    expect(isSame(instance.getValue(), createMoment('2018-01-01T00:30'))).toBe(true);

    instance.preMoment = null;
    instance.listItems.push({});
    instance.onListSelected([instance.listItems.length - 1]);
    expect(isSame(instance.getValue(), createMoment('1970-01-01T00:00'))).toBe(true);
  });

  test('misc', () => {
    wrapper = setup({ min: m2, max: m2 });
    let instance = wrapper.instance();

    wrapper.setProps({ value: m1 });
    expect(isSame(instance.getValue(), m2)).toBe(true);

    wrapper.setProps({ value: m3 });
    expect(isSame(instance.getValue(), m2)).toBe(true);

    wrapper = setup({ value: m2 });
    instance = wrapper.instance();
    wrapper.setProps({ value: createMoment('1800-01-01T00:00:00+00:00').utc() });
    expect(isSame(createMoment(new Date(1900, 0, 1)), instance.getValue())).toBe(true);
    wrapper.setProps({ value: createMoment(new Date()).add(60, 'year') });
    const systemMaxDate = Globalize.getToday().add(50, 'year');
    expect(instance.getValue().year()).toBe(systemMaxDate.year());
    expect(instance.getValue().month()).toBe(systemMaxDate.month());
    expect(instance.getValue().date()).toBe(systemMaxDate.date());

    wrapper.setProps({ value: m2 });
    instance.doSpin();
    expect(isSame(instance.getValue(), m1)).toBe(true);
    instance.doSpin(true);
    expect(isSame(instance.getValue(), m2)).toBe(true);

    jest.spyOn(instance.textProvider, 'isValid');
    instance.textProvider.isValid.mockImplementationOnce(() => false);
    expect(instance.initValue('0')).toBe(true);

    instance.textProvider.pattern = null;
    expect(instance.getTimeFormat()).toBe('hh:mm A');
  });
});
