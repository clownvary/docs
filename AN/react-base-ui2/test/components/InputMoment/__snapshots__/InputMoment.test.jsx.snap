// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/InputMoment/InputMoment.jsx rendering and basic useage 1`] = `
<InputMoment
  allowBlank={true}
  allowKeySpin={true}
  allowMouseSpin={false}
  ariaLabel="Input box"
  className=""
  currency="USD"
  data-qa-id=""
  disabled={false}
  flexibleCalendar={false}
  format="d"
  icon=""
  iconHint=""
  id=""
  items={Array []}
  listClassName=""
  listConfig={Object {}}
  listMaxHeight="300px"
  listPopupOptions={Object {}}
  listWidth="auto"
  name=""
  onBlur={[Function]}
  onCalendarClose={[Function]}
  onCalendarOpen={[Function]}
  onFocus={[Function]}
  onListClose={[Function]}
  onListOpen={[Function]}
  onTextChange={[Function]}
  onValueChange={[Function]}
  placeHolder=""
  prefix="an"
  showClear={false}
  showSpinner={false}
  showTrigger={false}
  showTrigger2={false}
  size="md"
  textAlign="left"
  timeStep={30}
  triggerIcon="icon-calendar-m"
  triggerIcon2="icon-clock-m"
  triggerIconHint=""
  triggerIconHint2=""
  triggerIconToggle=""
  triggerIconToggle2=""
  value={"2018-01-01T00:00:00.000Z"}
>
  <div
    className="an-input-moment input-group buttons-0 input-group--md"
  >
    <input
      aria-label="Input box"
      className="input input-group__field textalign-left input__field"
      data-qa-id=""
      disabled={false}
      id=""
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onDragEnter={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onMouseUp={[Function]}
      onPaste={[Function]}
      placeholder=""
      type="text"
    />
  </div>
</InputMoment>
`;
